### PR TITLE
add missing info to Makefile help

### PR DESCRIPTION
**Pull request by arthurus-rex**
Fill in missing Makefile help entries with brief descriptions, so all available make options are obvious.  Also created 'testing' section so users can run single testing tools if desired.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -104,11 +104,11 @@ base_dir := $(patsubst %/,%,$(dir $(realpath $(firstword $(MAKEFILE_LIST)))))
 kepler: build_containerized ## Build Kepler.
 .PHONY: kepler
 
-clean: clean-cross-build
+clean: clean-cross-build ## Undeploy Kepler.
```
Comment (maryamtahhan) [inline Makefile:107]:
```suggestion
clean: clean-cross-build ## Clean Kepler build
```

File: Makefile [hunk 2]
```diff
@@ -104,11 +104,11 @@ base_dir := $(patsubst %/,%,$(dir $(realpath $(firstword $(MAKEFILE_LIST)))))
...
 .PHONY: clean
 
 ##@ Container build.
-image_builder_check:
+image_builder_check:
```
Comment (maryamtahhan) [inline Makefile:111]:
```suggestion
image_builder_check:
```

File: Makefile [hunk 3]
```diff
@@ -221,21 +221,21 @@ containerized_build_container_rpm: ## Build the Containerized Kepler RPM inside
 		$(BUILDER_IMAGE) \
 		make build_container_rpm
 
-clean_build_local:
+clean_build_local: ## Undeploy Kepler for multiple archs.
```
Comment (maryamtahhan) [inline Makefile:224]:
```suggestion
clean_build_local: ## Clean local build directory
```

File: Makefile [hunk 4]
```diff
@@ -221,21 +221,21 @@ containerized_build_container_rpm: ## Build the Containerized Kepler RPM inside
...
 	rm -rf $(CROSS_BUILD_BINDIR)
 
-copy_build_local:
+copy_build_local: ## Copy bin to $(OUTPUT_DIR)/bin.
```
Comment (maryamtahhan) [inline Makefile:227]:
```suggestion
copy_build_local: ## Copy Kepler binary to $(OUTPUT_DIR)/bin.
```

File: Makefile [hunk 5]
```diff
@@ -307,20 +309,20 @@ bpf-test: generate ginkgo-set ## Run BPF tests.$(GOBIN)
 	$(SUDO) $(ENVTEST_ASSETS_DIR)/ginkgo \
 		./pkg/bpftest/bpftest.test
 
-escapes_detect: tidy-vendor
+escapes_detect: tidy-vendor ## Tidy Go dependencies.
```
Comment (maryamtahhan) [inline Makefile:312]:
```suggestion
escapes_detect: tidy-vendor 
```

I would move the description you added to the tidy-vendor target

File: Makefile [hunk 6]
```diff
@@ -390,13 +392,13 @@ dev: ## Setup development env using compose with 2 kepler (latest & current) dep
...
 
-dev-clean: ## Setup kepler (current and latest) along with
+dev-clean: ## Undeploy Kepler (current and latest) from development env.
 	docker compose \
 			-f $(COMPOSE_DIR)/$(DEV_TARGET)/compose.yaml \
 		down --remove-orphans --volumes --rmi all
 .PHONY: dev-clean
 
-dev-restart: dev-clean dev
+dev-restart: dev-clean dev ## Clean development environment.
```
Comment (maryamtahhan) [inline Makefile:401]:
```suggestion
dev-restart: dev-clean dev ## Restart development environment.
```

**Review by maryamtahhan — CHANGES_REQUESTED** [review]
Overall Looks good. There's a couple of comments in the wrong places.

**Review by maryamtahhan — APPROVED** [review]

**Review by sthaha — APPROVED** [review]

**Pull request merged**